components/camera/collectors_test.go

```diff
@@ -0,0 +1,208 @@
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/rimage"
+	tu "go.viam.com/rdk/testutils"
+	"go.viam.com/rdk/testutils/inject"
+	"go.viam.com/rdk/utils"
+)
+
+//nolint:lll
+var viamLogoJpegB64 = []byte("/9j/4QD4RXhpZgAATU0AKgAAAAgABwESAAMAAAABAAEAAAEaAAUAAAABAAAAYgEbAAUAAAABAAAAagEoAAMAAAABAAIAAAExAAIAAAAhAAAAcgITAAMAAAABAAEAAIdpAAQAAAABAAAAlAAAAAAAAABIAAAAAQAAAEgAAAABQWRvYmUgUGhvdG9zaG9wIDIzLjQgKE1hY2ludG9zaCkAAAAHkAAABwAAAAQwMjIxkQEABwAAAAQBAgMAoAAABwAAAAQwMTAwoAEAAwAAAAEAAQAAoAIABAAAAAEAAAAgoAMABAAAAAEAAAAgpAYAAwAAAAEAAAAAAAAAAAAA/9sAhAAcHBwcHBwwHBwwRDAwMERcRERERFx0XFxcXFx0jHR0dHR0dIyMjIyMjIyMqKioqKioxMTExMTc3Nzc3Nzc3NzcASIkJDg0OGA0NGDmnICc5ubm5ubm5ubm5ubm5ubm5ubm5ubm5ubm5ubm5ubm5ubm5ubm5ubm5ubm5ubm5ubm5ub/3QAEAAL/wAARCAAgACADASIAAhEBAxEB/8QBogAAAQUBAQEBAQEAAAAAAAAAAAECAwQFBgcICQoLEAACAQMDAgQDBQUEBAAAAX0BAgMABBEFEiExQQYTUWEHInEUMoGRoQgjQrHBFVLR8CQzYnKCCQoWFxgZGiUmJygpKjQ1Njc4OTpDREVGR0hJSlNUVVZXWFlaY2RlZmdoaWpzdHV2d3h5eoOEhYaHiImKkpOUlZaXmJmaoqOkpaanqKmqsrO0tba3uLm6wsPExcbHyMnK0tPU1dbX2Nna4eLj5OXm5+jp6vHy8/T19vf4+foBAAMBAQEBAQEBAQEAAAAAAAABAgMEBQYHCAkKCxEAAgECBAQDBAcFBAQAAQJ3AAECAxEEBSExBhJBUQdhcRMiMoEIFEKRobHBCSMzUvAVYnLRChYkNOEl8RcYGRomJygpKjU2Nzg5OkNERUZHSElKU1RVVldYWVpjZGVmZ2hpanN0dXZ3eHl6goOEhYaHiImKkpOUlZaXmJmaoqOkpaanqKmqsrO0tba3uLm6wsPExcbHyMnK0tPU1dbX2Nna4uPk5ebn6Onq8vP09fb3+Pn6/9oADAMBAAIRAxEAPwDm6K0dNu1tZsSgGNuDx0961NX09WT7ZbgcD5gPT1oA5qiul0fT1VPtlwByPlB7D1rL1K7W5mxEAI04GBjPvQB//9Dm66TRr/I+xTf8A/wrm6ASpBXgjpQB0ms34UfYof8AgWP5VzdBJY5PJNFAH//Z")
+
+const (
+	serviceName     = "camera"
+	captureInterval = time.Millisecond
+)
+
+func convertStringMapToAnyPBMap(params map[string]string) (map[string]*anypb.Any, error) {
+	methodParams := map[string]*anypb.Any{}
+	for key, paramVal := range params {
+		anyVal, err := convertStringToAnyPB(paramVal)
+		if err != nil {
+			return nil, err
+		}
+		methodParams[key] = anyVal
+	}
+	return methodParams, nil
+}
+
+func convertStringToAnyPB(str string) (*anypb.Any, error) {
+	var wrappedVal protoreflect.ProtoMessage
+	if boolVal, err := strconv.ParseBool(str); err == nil {
+		wrappedVal = wrapperspb.Bool(boolVal)
+	} else if int64Val, err := strconv.ParseInt(str, 10, 64); err == nil {
+		wrappedVal = wrapperspb.Int64(int64Val)
+	} else if uint64Val, err := strconv.ParseUint(str, 10, 64); err == nil {
+		wrappedVal = wrapperspb.UInt64(uint64Val)
+	} else if float64Val, err := strconv.ParseFloat(str, 64); err == nil {
+		wrappedVal = wrapperspb.Double(float64Val)
+	} else {
+		wrappedVal = wrapperspb.String(str)
+	}
+	anyVal, err := anypb.New(wrappedVal)
+	if err != nil {
+		return nil, err
+	}
+	return anyVal, nil
+}
+
+func TestCollectors(t *testing.T) {
+	logger := logging.NewTestLogger(t)
+	methodParams, err := convertStringMapToAnyPBMap(map[string]string{"camera_name": "camera-1", "mime_type": "image/jpeg"})
+	test.That(t, err, test.ShouldBeNil)
+	viamLogoJpeg, err := io.ReadAll(base64.NewDecoder(base64.StdEncoding, bytes.NewReader(viamLogoJpegB64)))
```

Comment:
q, is it actually important this is a valid jpeg, or can this be an arbitrary byte array?

Reply:
same question about `pcd`

Reply:
It is necessary for the images & pointclouds to be valid yes. This is b/c the Go SDK for camera (which the collector calls to get the data it is trying to capture):
https://github.com/viamrobotics/rdk/pull/4551/files/8694045a25f71f67bd774cebee5eec50889a63e8#diff-dcdd8001ea2dc2e271d7bbbb14f98041af0e66ffb025455461a246140cc0c5a3R182-R208

returns `image.Image` (part of the go standard library) and `pointcloud.Pointcloud` a type we define in rdk.

Technically speaking, we could shove invalid data into an image.Image, but I think it is less work to just test how we expect the software to work.

Reply:
gotcha, thank you for explaining 👍